### PR TITLE
[expo-permissions] Add rename `CAMERA_ROLL` to `MEDIA_LIBRARY` & add write only permission

### PR DESCRIPTION
**Pull request by lukmccall**
# Why

Since iOS 14, we can add write-only permission to the media library. Moreover, we want to remove `CAMERA_ROLL` to `MEDIA_LIBRARY`. 

# How

- `CAMERA_ROLL` -> `MEDIA_LIBRARY`
- added `WRITE_ONLY_MEDIA_LIBRARY`:
  - [x] media-library
  - [x] permissions
  - [x] image-picker
- updated documentation

# Test Plan

- NCL ✅

### CODE REVIEW

File: docs/pages/versions/unversioned/sdk/permissions.md [hunk 1]
```diff
@@ -52,23 +52,23 @@ Often you want to be able to test what happens when a user rejects a permission,
...
+| `USER_FACING_NOTIFICATIONS`                      | `expo-notifications`                                                                                                           |
+| `LOCATION`                                       | `expo-location`                                                                                                                |
+| `CAMERA`                                         | `expo-barcode-scanner` <br /> `expo-camera` <br /> `expo-face-detector` <br /> `expo-image-picker` <br /> `expo-media-library` |
+| `AUDIO_RECORDING`                                | `expo-av`                                                                                                                      |
+| `CONTACTS`                                       | `expo-contacts`                                                                                                                |
+| `MEDIA_LIBRARY`<br /> `WRITE_ONLY_MEDIA_LIBRARY` | `expo-image-picker`<br /> `expo-media-library`                                                                                 |
```
Comment (bbarthec) [inline docs/pages/versions/unversioned/sdk/permissions.md:67]:
```suggestion
| `MEDIA_LIBRARY`<br /> `MEDIA_LIBRARY_WRITE_ONLY` | `expo-image-picker`<br /> `expo-media-library`                                                                                 |
```
After a small inner struggle I believe that'a a better name for this permission 🤔

**Review by bbarthec — CHANGES_REQUESTED** [review]
That is looking good, but I prefer `MEDIA_LIBRARY_WRITE_ONLY` name for this permission 💅 I believe having `<context/subject>_<specifier>` is closely related to the idea of following [Node.JS error codes convention](https://nodejs.org/dist/latest-v14.x/docs/api/errors.html).

File: docs/pages/versions/unversioned/sdk/permissions.md [hunk 2]
```diff
@@ -276,7 +275,8 @@ interface PermissionInfo {
 - [`Permissions.CAMERA`](#permissionscamera) -- using the camera to capture images or videos
 - [`Permissions.AUDIO_RECORDING`](#permissionsaudio_recording) -- using the microphone to capture audio
 - [`Permissions.CONTACTS`](#permissionscontacts) -- reading or writing to contacts
-- [`Permissions.CAMERA_ROLL`](#permissionscamera_roll) -- accessing the images or videos from the user
+- [`Permissions.WRITE_ONLY_MEDIA_LIBRARY`](#permissionsmedia_library_write_only) -- writing to media library
```
Comment (bbarthec) [inline docs/pages/versions/unversioned/sdk/permissions.md:278]:
```suggestion
- [`Permissions.MEDIA_LIBRARY_WRITE_ONLY`](#permissionsmedia_library_write_only) -- writing to media library
```

File: packages/expo-permissions/src/Permissions.types.ts [hunk 3]
```diff
@@ -26,7 +28,7 @@ export interface PermissionMap {
 }
 
 export interface PermissionInfo extends UMPermissionResponse {
-  // iOS only - Permission.CAMERA_ROLL
+  // iOS only - Permission.MEDIA_LIBRARY/MEDIA_LIBRARY_WRITE_ONLY
```
Comment (bbarthec) [inline packages/expo-permissions/src/Permissions.types.ts:31]:
```suggestion
  /**
   * iOS only - Permission.MEDIA_LIBRARY/MEDIA_LIBRARY_WRITE_ONLY
   **/
```
~I'm not sure~ if non-block comments are applied to the signatures/variables and presented in VSCode 🤔 
_edit_: I've just confirmed that they're not 😉

**Review by bbarthec — APPROVED** [review]
LGTM 👍 
Found two small things 😉

**Comment by gregfenton** [comment]
Consider marking this as a "breaking change" for the SDK 40 release notes.

**Comment by KemaneWright** [comment]
Any updates on this?

**Comment by gregfenton** [comment]
> Any updates on this?

What update are you looking for?  The change is merged and was released with SDK 40.  My latest comment is about updating the SDK 40 release notes to reflect that this change requires existing apps to make adjustments to their code (i.e. this is a "breaking change").

**Comment by KemaneWright** [comment]
Oh sorry I meant to reference this thread https://github.com/MonstroDev/expo-image-picker-multiple/issues/30

For the image-picker-multiple package

**Pull request merged**